twill-api/src/main/java/org/apache/twill/api/TwillController.java

```diff
@@ -89,4 +90,44 @@
    * @return A {@link Future} that will be completed when the restart operation has been done.
    */
   Future<String> restartInstances(String runnable, int instanceId, int... moreInstanceIds);
+
```

Comment:
Still, the way I understand this is that it means, the message was received by the app master and passed on to the runnables. We have no idea whether the runnables have received or processed it. That can only be determined by examining the live node info for each runnable.